lib/src/async_map_buffer.dart

```diff
@@ -31,19 +31,21 @@ StreamTransformer<S, T> asyncMapBuffer<S, T>(
   var workFinished = StreamController()
     // Let the first event through.
     ..add(null);
-  return chainTransformers(
-      buffer(workFinished.stream), _asyncMapThen(convert, workFinished.add));
+  return chainTransformers(buffer(workFinished.stream),
+      _asyncMapThen(convert, () => workFinished.add(null)));
```

Comment:
Fwiw is going to be slower than the tearoff, so there is some negative value in this case.

Reply:
Good call. `void Function(void)` argument feels weird, but turns out to be fine for our use case if I go back to `.then` instead of `.whenComplete`.